services/remotecluster/service.go

```diff
@@ -49,6 +50,23 @@ type ServerIface interface {
 	GetMetrics() einterfaces.MetricsInterface
 }
 
+// RemoteClusterServiceIFace is used to allow mocking where a remote cluster service is used (for testing).
+// Unfortunately it lives here because the shared channel service, app layer, and server interface all need it.
+// Putting it in app layer means shared channel service must import app package.
```

Comment:
Could this still be added to `app` package if some types could be defined in `model` package? It's a good practice to define interfaces where they are _used_.

Reply:
I agree that is a good practise, however `model` is not where the interface is used either.  Given a choice between putting the interface where the implementation is versus a place where it is neither used nor implemented, I picked where it was implemented.

Reply:
Sorry I should've be more clear about this, I meant defining the interface in the `app` package but defining shared bits in the `model` package such as function types, structs etc.

Reply:
The interface is needed by shared channel service and app.  If defined in app then I need to import app into shared channel service which is not wanted.  Shared channel service uses a server interface that includes an API returning this interface, which is why we can't simply have two definitions of that interface.  It don't like it either, but it is an artifact of how server owns service instances and how we can mock server and those instances.  BTW, there is precedent for breaking the interface where it is used guideline in the standard lib.

Reply:
I see, thanks for the clarification.